src/main/scala/com/github/saurfang/sas/mapred/SasRecordReader.scala

```diff
@@ -27,131 +27,157 @@ import org.apache.hadoop.mapred.{FileSplit, InputSplit, RecordReader}
 import org.apache.log4j.LogManager
 
 /**
- * An [[RecordReader]] for [[SasInputFormat]]. Each split is aligned to page sized specified in the .sas7bdat meta info.
- * Each split looks back previous split if the start of split is incomplete.
+ * An [[RecordReader]] for [[SasInputFormat]].
+ * Each split is aligned to the closest preceding page boundary, calculated from the page size specified in the .sas7bdat meta info.
  */
 class SasRecordReader(job: Configuration, split: InputSplit) extends RecordReader[NullWritable, Array[Object]] {
-
-  private var recordCount: Int = 0
-  private var splitStart: Long = 0L
-  private var splitEnd: Long = 0L
-  private var lastPageBlockCounter: Int = 0
-  private var recordValue: Array[Object] = _
-
+  
   @transient lazy val log = LogManager.getLogger(this.getClass.getName)
-
-  // the file input
+    
+  //// Warm-ups 
+  // Process InputSplit
   private val fileSplit = split.asInstanceOf[FileSplit]
-
-  // the byte position this fileSplit starts at
-  splitStart = fileSplit.getStart
-
-  // splitEnd byte marker that the fileSplit ends at
-  splitEnd = splitStart + fileSplit.getLength
-
-  // the actual file we will be reading from
-  private val file = fileSplit.getPath
-  // check compression
-  private val codec = new CompressionCodecFactory(job).getCodec(file)
+  private val filePath = fileSplit.getPath
+  
+  // Sanity Check: Ensure file is not compressed.
+  private val codec = new CompressionCodecFactory(job).getCodec(filePath)
   if (codec != null) {
-    throw new IOException("SASRecordReader does not support reading compressed files")
+    throw new IOException("SASRecordReader does not support reading compressed files.")
   }
-  // get the filesystem
-  private val fs = file.getFileSystem(job)
-  // open the File
-  private val fileInputStream = fs.open(file)
+  
+  
+  //// Initialize
+  // Variables
+  private var recordCount: Int = 0
```

Comment:
Is this records per split? or per file.. if it's per file or if file is not splittable, `long` datatype might be better IMO

> "The maximum size of a SAS data set in a Direct Access Bound Library is limited by the maximum size of the library, which is about 2986 GB on 3390 volumes."
> 
> http://support.sas.com/kb/8/213.html
> 
>

Reply:
Agreed